src/mlpack/core/metrics/non_maximal_supression_impl.hpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file nms_metric_impl.hpp
+ * @author Kartik Dutt
+ *
+ * Implementation of Non Maximal Supression metric.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_METRICS_NMS_IMPL_HPP
+#define MLPACK_CORE_METRICS_NMS_IMPL_HPP
+
+// In case it hasn't been included.
+#include "non_maximal_supression.hpp"
+
+namespace mlpack {
+namespace metric {
+
+template<bool UseCoordinates, bool Transpose>
+template<
+    typename BoundingBoxesType,
+    typename ConfidenceScoreType,
+    typename OutputType
+>
+void NMS<UseCoordinates, Transpose>::Evaluate(
+    const BoundingBoxesType& boundingBoxes,
+    const ConfidenceScoreType& confidenceScores,
+    OutputType& selectedBoundingBoxes,
+    const double threshold)
+{
+  Log::Assert(boundingBoxes.n_rows == 4, "Bounding boxes must be \
+      must contain only 4 rows determining coordinates of bounding \
+      box either in {x1, y1, x2, y2} or {x1, y1, h, w} format.\
+      Refer to the documentation for more information.");
+  // Clear selected bounding boxes.
+  selectedBoundingBoxes.clear();
+  // Obtain Sorted indices for bounding boxes according to
+  // their confidence scores.
+  arma::uvec sortedIndices = arma::sort_index(confidenceScores);
+
+  // Pre-Compute area of each bounding box.
+  arma::mat area;
+  if (UseCoordinates)
+  {
+    area = (boundingBoxes.row(2) - boundingBoxes.row(0)) %
+        (boundingBoxes.row(3) - boundingBoxes.row(1));
+  }
+  else
+  {
+    area = (boundingBoxes.row(2)) % (boundingBoxes.row(3));
+  }
+
+  while (sortedIndices.n_elem > 0)
+  {
+    size_t selectedIndex = sortedIndices(sortedIndices.n_elem - 1);
+    // Choose the box with the largest probability.
+    selectedBoundingBoxes.insert_cols(0, boundingBoxes.col(selectedIndex));
+    // Check if there are other bounding boxes to compare with.
+
+    if (sortedIndices.n_elem == 1)
+    {
+      break;
+    }
```

Comment:
Can you give a space between the code and comment?
```suggestion
    size_t selectedIndex = sortedIndices(sortedIndices.n_elem - 1);
    
    // Choose the box with the largest probability.
    selectedBoundingBoxes.insert_cols(0, boundingBoxes.col(selectedIndex));
    
    // Check if there are other bounding boxes to compare with.
    if (sortedIndices.n_elem == 1)
    {
      break;
    }
```